### PR TITLE
DAOS-6588 restore Requires for devel/tests packages

### PR DESCRIPTION
**Pull request by kccain**
Signed-off-by: Kenneth Cain <kenneth.c.cain@intel.com>

### CODE REVIEW

File: hdf5-vol-daos.spec [hunk 1]
```diff
@@ -72,6 +72,7 @@ HDF5 VOL DAOS with MPICH
 Summary: HDF5 VOL DAOS devel with MPICH
 BuildRequires: hdf5-mpich-devel%{?_isa}
 Requires: hdf5-mpich-devel%{?_isa}
+Requires: %{name}-mpich2
```
Comment (brianjmurrell) [inline hdf5-vol-daos.spec:75]:
```suggestion
Requires: %{name}-mpich2%{?_isa} = %{version}-%{release}
```
is the correct syntax.

File: hdf5-vol-daos.spec [hunk 2]
```diff
@@ -80,6 +81,7 @@ HDF5 VOL DAOS devel with MPICH
 %package mpich-tests
 Summary: HDF5 VOL DAOS tests with mpich
 BuildRequires: hdf5-mpich-devel%{?_isa}
+Requires: %{name}-mpich2
```
Comment (brianjmurrell) [inline hdf5-vol-daos.spec:84]: Ditto.

File: hdf5-vol-daos.spec [hunk 3]
```diff
@@ -101,6 +103,7 @@ HDF5 VOL DAOS with OpenMPI 3
 Summary: HDF5 VOL DAOS devel with OpenMPI 3
 BuildRequires: hdf5-openmpi3-devel%{?_isa}
 Requires: hdf5-openmpi3-devel%{?_isa}
+Requires: %{name}-openmpi3
```
Comment (brianjmurrell) [inline hdf5-vol-daos.spec:106]: Ditto.

File: hdf5-vol-daos.spec [hunk 4]
```diff
@@ -109,6 +112,7 @@ HDF5 VOL DAOS devel with OpenMPI 3
 %package openmpi3-tests
 Summary: HDF5 VOL DAOS tests with openmpi3
 BuildRequires: hdf5-openmpi3-devel%{?_isa}
+Requires: %{name}-openmpi3
```
Comment (brianjmurrell) [inline hdf5-vol-daos.spec:115]: Ditto.

**Review by brianjmurrell — CHANGES_REQUESTED** [review]

**Review by brianjmurrell — APPROVED** [review]

**Comment by brianjmurrell** [comment]
> ￼ kccain requested a review from daos-stack/release-engineering 26 minutes ago

@kccain Testing with this PR on daos@master proved it resolved the issue there?

**Comment by kccain** [comment]
> > ￼ kccain requested a review from daos-stack/release-engineering 26 minutes ago
> 
> @kccain Testing with this PR on daos@master proved it resolved the issue there?

Yes for the io/daos_vol.py problem (openmpi related test cases) reported in DAOS-6588 a PR based on daos master was created and run with the packages generated by this PR. All cases either were skipped or passed.
https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-4387/2/testReport/(root)/DaosVol/

a successful test run set of logs has also been attached to DAOS-6588
https://jira.hpdd.intel.com/secure/attachment/33978/33978_job-2021-01-23T07.50-ed372d6-io-daos_vol.zip

**Pull request merged**